ext/bg/js/backend.js

```diff
@@ -212,22 +213,35 @@ class Backend {
     }
 
     getProfile(optionsContext) {
-        const profiles = this.options.profiles;
+        return this._getProfilesFromContext(optionsContext).next().value.profile;
+    }
+
+    getMatchingProfiles(optionsContext) {
+        return Array.from(this._getProfilesFromContext(optionsContext));
+    }
+
+    *_getProfilesFromContext(optionsContext) {
+        const {profiles, profileCurrent} = this.options;
+
         if (typeof optionsContext.index === 'number') {
-            return profiles[optionsContext.index];
+            yield {index: optionsContext.index, profile: profiles[optionsContext.index]};
+            return;
         }
-        const profile = this.getProfileFromContext(optionsContext);
-        return profile !== null ? profile : this.options.profiles[this.options.profileCurrent];
-    }
 
-    getProfileFromContext(optionsContext) {
-        for (const profile of this.options.profiles) {
+        let contextHasResults = false;
+        let profileIndex = 0;
+        for (const profile of profiles) {
             const conditionGroups = profile.conditionGroups;
             if (conditionGroups.length > 0 && Backend.testConditionGroups(conditionGroups, optionsContext)) {
-                return profile;
+                contextHasResults = true;
+                yield {index: profileIndex, profile};
             }
+            ++profileIndex;
+        }
+
+        if (!contextHasResults) {
+            yield {index: profileCurrent, profile: profiles[profileCurrent]};
         }
```

Comment:
Could be useful to yield all profiles but also indicate which is the currently selected profile in case the user has not specified any conditions or none of them match the current context

Reply:
Fixed in https://github.com/FooSoft/yomichan/pull/389/commits/5d93d0d853618dc9537327a85577f10cc9ac5457